tests/components/deako/test_light.py

```diff
@@ -0,0 +1,259 @@
+
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components.deako.const import DOMAIN
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    DOMAIN as LIGHT_DOMAIN,
+    ColorMode,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.asyncio
+async def test_light_setup_no_devices(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test light platform setup with no devices returned."""
+    mock_config_entry.add_to_hass(hass)
+
+    # no devices
+    pydeako_deako_mock.get_devices.return_value = {}
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+
+@pytest.mark.asyncio
+async def test_light_setup_with_device(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test light platform setup with device returned."""
+    mock_config_entry.add_to_hass(hass)
+
+    pydeako_deako_mock.return_value.get_devices.return_value = {
+        "some_device": {},
+    }
+    pydeako_deako_mock.return_value.get_name.return_value = "some device"
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_entry = entity_registry.async_get("light.some_device")
+    assert entity_entry
+
+
+@pytest.mark.asyncio
+async def test_light_initial_props(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    pydeako_deako_mock: MagicMock,
+    entity_registry: er.EntityRegistry,
+    device_registry: dr.DeviceRegistry,
+) -> None:
+    """Test on/off light is setup with accurate initial properties."""
+    mock_config_entry.add_to_hass(hass)
+
+    pydeako_deako_mock.return_value.get_devices.return_value = {
+        "uuid": {
+            "name": "kitchen",
+        }
+    }
+    pydeako_deako_mock.return_value.get_name.return_value = "kitchen"
+    pydeako_deako_mock.return_value.get_state.return_value = {
+        "power": False,
+    }
+    pydeako_deako_mock.return_value.is_dimmable.return_value = False
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # make sure our initial state exists
+    assert (state := hass.states.get(f"{LIGHT_DOMAIN}.kitchen"))
+    assert state.state == STATE_OFF
+
+    # test other properties
+    entity = entity_registry.async_get(f"{LIGHT_DOMAIN}.kitchen")
+    assert entity
+    assert entity.unique_id == "uuid"
+    assert entity.original_name is None
+    assert ColorMode.ONOFF in entity.capabilities.get("supported_color_modes")
```

Comment:
You can look into snapshotting, use snapshot_platform to make sure you snapshot the entity and the state